spec/services/induction/amend_participant_cohort_spec.rb

```diff
@@ -377,6 +383,44 @@
             end
           end
         end
+
+        context "when source_cohort_start_year matches target_cohort_start_year" do
+          subject(:form) do
+            described_class.new(participant_profile:,
+                                source_cohort_start_year: target_cohort_start_year,
+                                target_cohort_start_year:)
+          end
+
+          before do
+            participant_profile.update!(schedule: create(:ecf_extended_schedule, cohort: target_cohort))
+            Induction::ChangeProgramme.call(participant_profile:,
+                                            end_date: Date.current,
+                                            new_induction_programme: target_school_cohort.default_induction_programme,
+                                            start_date: Date.current)
+          end
+
+          it "returns true and set no errors" do
+            expect(form.save).to be_truthy
+            expect(form.errors).to be_empty
+          end
+
+          it "keeps the participant in the same schedule and cohort" do
+            current_induction_record = participant_profile.induction_records.latest
+            historical_induction_record = (participant_profile.induction_records - [current_induction_record]).first
+
+            expect { form.save }.to not_change { current_induction_record.reload.schedule }
+                                      .and not_change { current_induction_record.reload.cohort_start_year }
+                                             .and not_change { current_induction_record.reload.induction_programme }
+                                                    .and not_change { current_induction_record.reload.participant_profile.schedule }
+                                                           .and not_change { current_induction_record.reload.participant_profile.school_cohort }
+                                                                  .and change { historical_induction_record.reload.schedule }
+                                                                         .from(historical_induction_record.schedule)
+                                                                         .to(current_induction_record.schedule)
+                                                                         .and change { historical_induction_record.reload.cohort_start_year }
+                                                                                .from(historical_induction_record.cohort_start_year)
+                                                                                .to(current_induction_record.cohort_start_year)
```

Comment:
Normally I'd say there's a lot going on here and suggest it's split into separate expectations, but if it's urgent let's get it shipped and revisit later?

Reply:
Thanks Peter. You are right. I just wanted to not make the suite too slow but the test looks quite unreadable and not clear what's tested. 
Refactored!.